File: fundamentos/desafio-dart/lib/domains/account.dart
class Account {
  final int id;
  final int agencia;
  final int conta;
  final String name;
  final double balance;

  Account(
      {this.id, this.agencia, this.conta, this.name = '', this.balance = 0});

  Account copyWith({int agencia, int conta, String name, double balance}) {
    return Account(
        agencia: agencia ?? this.agencia,
        conta: conta ?? this.conta,
        name: name ?? this.name,
        balance: balance ?? this.balance);
  }

  Account.fromJson(Map<String, dynamic> json)
      : id = json['id'],
        agencia = json['agencia'],
        conta = json['conta'],
        name = json['name'],
        balance = json['balance'] + .0; // +.0 cast possible int to double
}

File: fundamentos/desafio-java/src/main/java/DesafioJava/Domain/Account.java
package DesafioJava.Domain;

public class Account {
    
    Integer id;
    Integer agencia;
    Integer conta;
    String name;
    Integer balance = 0;
    
    public Account() {}
    
    public Account(Integer id,
	           Integer agencia,
	           Integer conta,
	           String name,
	           Integer balance) {
	setId(id);
	setAgencia(agencia);
	setConta(conta);
	setName(name);
	setBalance(balance);
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public Integer getAgencia() {
        return agencia;
    }

    public void setAgencia(Integer agencia) {
        this.agencia = agencia;
    }

    public Integer getConta() {
        return conta;
    }

    public void setConta(Integer conta) {
        this.conta = conta;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Integer getBalance() {
        return balance;
    }

    public void setBalance(Integer balance) {
        this.balance = balance;
    }

}
